### PR TITLE
[ISSUE33]Add extension Attributes & Interface

### PR DESCRIPTION
**Pull request by duhenglucky**
(1) Add extension  attributes & interface.
(2) Keep consistent with specification.

### CODE REVIEW

File: openmessaging-api/src/main/java/io/openmessaging/extension/ExtensionClient.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * <p>
+ * This interface contains some methods are used for getting configurations related implementation. but this interface
+ * are not mandatory.
+ * </p>
+ *
+ * @version OMS 1.0.0
+ * @since OMS 1.0.0
+ */
+public interface ExtensionClient {
```
Comment (liyue2008) [inline openmessaging-api/src/main/java/io/openmessaging/extension/ExtensionClient.java]: Is there any way to get a instance for the new added interface io.openmessaging.extension.ExtensionClient.

There should be a getExcensionCleint method in the MessagingAccessPoint or other interfaces.

Comment (duhenglucky) [reply openmessaging-api/src/main/java/io/openmessaging/extension/ExtensionClient.java]: yep, I also think it should be a method which can be found in Producer or Consumer. what's your opinion?

File: openmessaging-api/src/main/java/io/openmessaging/KeyValue.java [hunk 2]
```diff
@@ -82,18 +82,8 @@
      * @return the value in this {@code KeyValue} object with the specified key value
      * @see #put(String, short)
      */
-    int getShort(String key);
+    short getShort(String key);
 
-    /**
```
[the remark this answers is not in the document]
Comment (duhenglucky) [reply openmessaging-api/src/main/java/io/openmessaging/KeyValue.java]: It seems that we should define so many interface or method?

[the remark this answers is not in the document]
Comment (liyue2008) [reply openmessaging-api/src/main/java/io/openmessaging/KeyValue.java]: No need， just repliace  the method "**short** getShort(String key);" to "**Short** getShort(String key);"

File: openmessaging-api/src/main/java/io/openmessaging/extension/ExtensionClient.java (hunk 1 above)
Comment (liyue2008) [reply openmessaging-api/src/main/java/io/openmessaging/extension/ExtensionClient.java]: Aggree that.

**Pull request merged**